tools/unused.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import b
+
+# @run_on_docker
+def register(command):
+    # Install help
+    command.help = "Creates a list of unused or commented out modules"
+
+
+# @run_on_docker
+def run(**kwargs):
+
+    source_path = b.cmake_cache["NUbots_SOURCE_DIR"]
+    roles_path = source_path + "/roles"
+    modules_path = source_path + "/modules"
+    # print(roles_path, modules_path)
+    existing_modules = {}
+    missing_modules = {}
+    commented_modules = {}
+    used_modules = []
+    unused_commented_modules = []
+
+    # a list of folders that could be within a module
+    skip_list = ["src", "data", "config", "tests", "report"]
+    skip_list = [s + os.sep for s in skip_list]
```

Comment:
No need to do a skip list, Just search every single folder for one that matches what a module would look like
```python
re.search(r'(?:[^/]+/)*(\w+)/src/(\w+).hp{2}?$')`
```

Reply:
Previously you had suggested the following:

> A better way than using a skip list would be to look for the CMakeLists.txt file in the directory and finding a NUClearModule call inside it.

> Rather than doing it this way, just search for files named CMakeLists.txt that contain a call to NUClearModule(...)

Reply:
That's a way better idea! Good job past me and bad present me for forgetting...